src/rust/aggregator/src/util.rs

```diff
@@ -0,0 +1,27 @@
+use anyhow::anyhow;
+use sqlx::{Executor, Pool, Transaction};
+use sqlx_postgres::Postgres;
+
+use crate::{PipelineAggregationResult, PipelineError};
+
+pub fn to_pipeline_error<T: Into<anyhow::Error>>(e: T) -> PipelineError {
+    PipelineError::ProcessingError(anyhow!(e))
+}
+
+pub async fn create_repeatable_read_transaction<'a>(
+    pool: &Pool<Postgres>,
+) -> Result<Transaction<'a, Postgres>, PipelineError> {
+    let mut transaction = pool.begin().await.map_err(to_pipeline_error)?;
+    transaction
+        .execute("SET TRANSACTION ISOLATION LEVEL REPEATABLE READ;")
+        .await
+        .map_err(to_pipeline_error)?;
+    Ok(transaction)
+}
+
+pub async fn commit_transaction<'a>(tx: Transaction<'a, Postgres>) -> PipelineAggregationResult {
```

Comment:
Note that repeatable read can still have serialization errors: Repeatable read can still through txn errors: https://www.postgresql.org/docs/current/transaction-iso.html#XACT-REPEATABLE-READ

If that's the case, how about returning an error to the caller?

Then, depending on the pipeline, either gracefully igoring or otherwise

Reply:
Alternatively it could retry, like in the processor

Reply:
As noted above, what happened to us with the weird "maybe it will succeed if retried" error, it is not when commiting that the error occurs, but on individual calls.

Reply:
Noting that the error is from a query, not form the commit operation

Also, this is handled by using a container with auto-restart